contrib/pkg/broker/server/server_test.go

```diff
@@ -117,12 +119,12 @@ func readJSON(rr *httptest.ResponseRecorder) (map[string]interface{}, error) {
 type Controller struct {
 	t *testing.T
 
-	catalog               func() (*brokerapi.Catalog, error)
-	getServiceInstance    func(id string) (string, error)
-	createServiceInstance func(id string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, error)
-	removeServiceInstance func(id string) (*brokerapi.DeleteServiceInstanceResponse, error)
-	bind                  func(instanceID string, bindingID string, req *brokerapi.BindingRequest) (*brokerapi.CreateServiceBindingResponse, error)
-	unBind                func(instanceID string, bindingID string) error
+	catalog                         func() (*brokerapi.Catalog, error)
+	getServiceInstanceLastOperation func(id string) (*brokerapi.LastOperationResponse, error)
```

Comment:
@vaikas-google which broker are you talking about? This controller is just a stub with delegate methods for unit testing of OSB API server. This test currently covers only `/v2/catalog` endpoint, but we should cover all REST API methods in the future.

If you are referring to [user_provided/.../controller.go](https://github.com/kubernetes-incubator/service-catalog/blob/master/contrib/pkg/broker/user_provided/controller/controller.go), it's just one of the controller implementations, nothing stops us from implementing `user_provided_async/.../controller.go` in the future.